edkrepo/commands/list_repos_command.py

```diff
@@ -45,92 +47,117 @@ def get_metadata(self):
+                        for source in sources:
+                            repo_urls.add(self.get_repo_url(source.remote_url))
+            for manifest_repo in user_config_manifest_repos:
+                # Get path to global manifest file
+                global_manifest_directory = config['user_cfg_file'].manifest_repo_abs_path(manifest_repo)
+                if args.verbose:
+                    print(humble.MANIFEST_DIRECTORY)
+                    print(global_manifest_directory)
+                    print()
+                #Create a dictionary containing all the manifests listed in the CiIndex.xml file
+                index_path = os.path.join(global_manifest_directory, CI_INDEX_FILE_NAME)
+                ci_index_xml = CiIndexXml(index_path)
+                user_config_manifest_repos_project_list = ci_index_xml.project_list
                 if args.archived:
-                    combo_list.extend([c.name for c in manifest.archived_combinations])
-                for combo in combo_list:
-                    sources = manifest.get_repo_sources(combo)
-                    for source in sources:
-                        repo_urls.add(self.get_repo_url(source.remote_url))
+                    user_config_manifest_repos_project_list.extend(ci_index_xml.archived_project_list)
+                for project in user_config_manifest_repos_project_list:
+                    xml_file = ci_index_xml.get_project_xml(project)
+                    manifest = ManifestXml(os.path.normpath(os.path.join(global_manifest_directory, xml_file)))
+                    found_manifests['{}:{}'.format(manifest_repo, project)] = manifest
+                    combo_list = [c.name for c in manifest.combinations]
+                    if args.archived:
+                        combo_list.extend([c.name for c in manifest.archived_combinations])
+                    for combo in combo_list:
+                        sources = manifest.get_repo_sources(combo)
+                        for source in sources:
+                            repo_urls.add(self.get_repo_url(source.remote_url))
 
-        #Remove the manifest repo portion of the key is there is not a duplicate project name
-        key_list = list(found_manifests)
-        for entry in key_list:
-            new_key = entry.split(':')[1]
-            value = found_manifests[entry]
-            del found_manifests[entry]
-            for found_manifest in list(found_manifests):
-                if found_manifest.split(':')[1] == new_key:
-                    new_key = 'Manifest Repository: {} Project: {}'.format(entry.split(':')[0], entry.split(':')[1])
-                    break
-            if new_key in manifests.keys():
-                new_key = 'Manifest Repository: {} Project: {}'.format(entry.split(':'[0]), entry.split(':')[1])
-            manifests[new_key] = value
+            #Remove the manifest repo portion of the key is there is not a duplicate project name

```

Comment:
The phrasing of this comment is confusing. Should it be 'remove the manifest repo portion of the key so there is not a duplicate project name'